internal/services/containerapps/container_app_environment_resource.go

```diff
@@ -424,24 +442,31 @@ func (r ContainerAppEnvironmentResource) Update() sdk.ResourceFunc {
 			}
 
 			if metadata.ResourceData.HasChanges("logs_destination", "log_analytics_workspace_id") {
+				// For 4.x we need to be compensate for the legacy behaviour of setting log destination based on the presence of log_analytics_workspace_id
+				if !features.FivePointOhBeta() && metadata.ResourceData.GetRawConfig().AsValueMap()["logs_destination"].IsNull() && state.LogAnalyticsWorkspaceId == "" {
```

Comment:
@sreallymatt has a PR open to remove the `FivePointOhBeta` flag and all uses of it, please use `FivePointOh` going forward
```suggestion
				if !features.FivePointOh() && metadata.ResourceData.GetRawConfig().AsValueMap()["logs_destination"].IsNull() && state.LogAnalyticsWorkspaceId == "" {
```